pingcap/dm
# make: add check for merge conflicts

### What problem does this PR solve? 
ref https://github.com/pingcap/ticdc/pull/2339/files

fail the CI when detect git merge conflicts

### What is changed and how it works?

### Check List 

Tests 

 - Manual test

```
$ make check
check-merge-conflicts
.github/workflows/binlog-999999.yml:<<<<<<< HEAD
...
make: *** [check_merge_conflicts] Error 1
```


Related changes

 - Need to cherry-pick to the release branch
